lib/pyseq/logger.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2024, Ryan Galloway (ryan@rsgalloway.com)
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#  - Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+#  - Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+#  - Neither the name of the software nor the names of its contributors
+#    may be used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+__doc__ = """
+Contains logging functionality for the pyseq module. By default, this creates
+a null handler so that importing application can set up their own stream hanlder.
+"""
+
+import logging
+import os
+
+LOG_LEVEL = int(os.environ.get("LOG_LEVEL", 20))
+
+log = logging.Logger("pyseq")
+log.setLevel(LOG_LEVEL)
+log.addHandler(logging.NullHandler())
+
+
+def setup_stream_handler():
+    """Adds a new stdout stream handler."""
+    for h in log.handlers:
+        if h.name == log.name and "StreamHandler" in str(h):
+            del log.handlers[log.handlers.index(h)]
```

Comment:
ah, cool, don't think I knew that. what about just adding a format string that includes the module name? e.g.

``` python
"[%(asctime)s] - %(module)10s:%(lineno)3d - %(levelname)-7s - %(message)s"
```

Reply:
If this was in response to the naming loggers from the module, yeah that works too.  I was just pointing out the module has some built-in functionality for organising loggers easily.  It's nice if you take advantage of it, but with a small package, and if you're not including the logger name in the records, then it may not be worth it.

Reply:
it's not really even logging at all (I think there are only 2 calls to the logger total, both in seq.py). I wonder if the logging should just be removed and use print statements instead. 

> I think one of my comments does need to be addressed

just to be sure I address it, which comment did you mean?

Reply:
I think generally logging is preferred over prints, but it's up to you.
The thing to address was the deleting an item from the `log.handlers` while iterating over it, which it looks like you've removed.  So it's no longer valid.

Reply:
Good enough for me.

Reply:
thanks for taking the time to review ;)